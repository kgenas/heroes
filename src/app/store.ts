import {
  combineReducers,
  configureStore,
  PreloadedState,
  ThunkAction,
  Action
} from '@reduxjs/toolkit'
import heroReducer from '../features/counter/loginHero';
// Create the root reducer independently to obtain the RootState type
const rootReducer = combineReducers({
    hero: heroReducer
})

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState
  })
}
export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
