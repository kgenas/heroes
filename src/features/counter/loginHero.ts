import { createSlice, PayloadAction } from '@reduxjs/toolkit';

export interface HeroState {
  name: string;
  logged?: boolean;
}

const initialState: HeroState = {
  name: '',
  logged:false,
};

export const hero = createSlice({
  name: 'hero',
  initialState,
  reducers: {
    login: (state , action: PayloadAction<HeroState>) => {
      const { name } = action.payload;
      return{
        ...state,
        logged: true,
        name
      };
    },
    logout: (state) => {
      state.logged = initialState.logged;
    },    
  },  
});

export const { login , logout } = hero.actions;
export default hero.reducer;
