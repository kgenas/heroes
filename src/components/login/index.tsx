import React from "react";
import { useNavigate } from "react-router-dom"
import { useDispatch } from 'react-redux';
import { AppDispatch } from '../../app/store';
import { login } from "../../features/counter/loginHero";

export const Login = () =>{
    const navigate = useNavigate();
    const dispatch:AppDispatch = useDispatch();

    const handleLogion = () =>{

        const data = {
            name: 'pedro',                
        };
        dispatch(login(data));
        const lastPath = localStorage.getItem('lasPath') || '/marvel';
        navigate( lastPath, { replace: true });
    };
    return(
        <div className="container mt-5">
            <h1>Login</h1>
            <hr/>
            <button
                className="btn btn-primary"
                onClick={ handleLogion }
            >
                Login
            </button>
        </div>
    )    
}