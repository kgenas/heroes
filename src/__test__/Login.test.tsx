import React from 'react';
import { fireEvent, screen } from "@testing-library/react";
import { setupStore } from '../app/store';
import { login } from '../features/counter/loginHero';
import { renderWithProviders } from '../utils/test-utils';
import { Login } from '../components/login/index';

describe('Login',()=>{
    test('Click',()=>{
        const store = setupStore();
        store.dispatch(login({
            name:'Pedro'
        }))                   
        renderWithProviders(<Login/>,{ store });     
        //eslint-disable-next-line 
        fireEvent.click(screen.getAllByText(/Login/)[1])
        //eslint-disable-next-line    
        screen.debug();
    })
    test('Login Snapshot',() =>{
        const { baseElement } = renderWithProviders(<Login/>);
        expect( baseElement ).toMatchSnapshot();
      })
})