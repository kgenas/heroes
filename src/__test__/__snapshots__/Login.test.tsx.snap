// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Login Snapshot 1`] = `
<body>
  <div>
    <div
      class="container mt-5"
    >
      <h1>
        Login
      </h1>
      <hr />
      <button
        class="btn btn-primary"
      >
        Login
      </button>
    </div>
  </div>
</body>
`;
